probe-rs/src/architecture/arm/communication_interface.rs

```diff
@@ -3,16 +3,48 @@ use super::{
         valid_access_ports, APAccess, APClass, APRegister, AccessPort, BaseaddrFormat, GenericAP,
         MemoryAP, BASE, BASE2, IDR,
     },
-    dp::Select,
+    dp::{Abort, Ctrl, DPAccess, DPRegister, DPv1, DebugPort, DebugPortId, Select, DPIDR},
     memory::romtable::{CSComponent, CSComponentId, PeripheralID},
     memory::ADIMemoryInterface,
 };
 use crate::config::ChipInfo;
-use crate::{CommunicationInterface, DebugProbe, DebugProbeError, Error, Memory, Probe};
+use crate::{
+    CommunicationInterface, DebugProbe, DebugProbeError, Error as ProbeRsError, Memory, Probe,
+};
 use jep106::JEP106Code;
-use log::debug;
 use std::cell::RefCell;
 use std::rc::Rc;
+use thiserror::Error;
+
+#[derive(Debug, Error)]
+pub enum DapError {
+    #[error("Unexpected answer to command")]
+    UnexpectedAnswer,
+    #[error("DAP Error")]
+    DAP,
+    // TODO: Remove
+    #[error("Too much data provided for SWJ Sequence command")]
+    TooMuchData,
+    // TODO: Remove
+    #[error("Error in the USB HID access")]
```

Comment:
Please also include the wrapped error in the print output, that helps with debugging:
```suggestion
    #[error("Error in the USB HID access: {0}")]
```